frontend/src/components/AgencyProfile/AgencySideBar.jsx

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import { Component, useEffect, useState } from "react";
+import "./AgencyProfile.css";
+import { NavHashLink } from "react-router-hash-link";
+
+/**
+ * Functional component for Agency Side Bar
+ *
+ * @param {*} getScrollPositions function that returns page offset of agency categories
+ * @param {string} id unique id of agency
+ * @param {*} ScrollTo function that scrolls page to given element
+ * @returns {*} Agency Side Bar Component
+ */
+function AgencySideBar({ getScrollPositions, id, positions, ScrollTo }) {
+  const [scrollPos, setScrollPos] = useState("");
+
+  const resetSideBar = () => {
+    if (document.getElementById("main") !== null) {
+      document.getElementById("main").className = "header-side-container";
+    }
+    if (document.getElementById("contacts") !== null) {
+      document.getElementById("contacts").className = "header-side-container";
+    }
+    if (document.getElementById("capacity") !== null) {
+      document.getElementById("capacity").className = "header-side-container";
+    }
+    if (document.getElementById("compliance") !== null) {
+      document.getElementById("compliance").className = "header-side-container";
+    }
+    if (document.getElementById("demographics") !== null) {
+      document.getElementById("demographics").className =
+        "header-side-container";
+    }
+    if (document.getElementById("retail") !== null) {
+      document.getElementById("retail").className = "header-side-container";
+    }
+    if (document.getElementById("tasks") !== null) {
+      document.getElementById("tasks").className = "header-side-container";
+    }
+  };
+
+  const handleScroll = () => {
+    let positions = getScrollPositions();
+    switch (true) {
+      case window.innerHeight + window.pageYOffset >=
+        document.body.offsetHeight:
+        resetSideBar();
+        if (document.getElementById("tasks") !== null) {
+          document.getElementById("tasks").className =
+            "header-side-active-container";
+        }
+        setScrollPos("tasks");
+        break;
+      case positions[5] <= 160:
```

Comment:
Same with here- what is the importance of 160? Can this be made a constant as well?